File: Stack/python/lib/stack.py
class Stack:
    def __init__(self):
        self.items = []

    def is_empty(self):
        return len(self.items) == 0

    def push(self, item):
        self.items.append(item)

    def pop(self):
        if not self.is_empty():
            return self.items.pop()
        else:
            raise IndexError("Stack is empty")

    def peek(self):
        if not self.is_empty():
            return self.items[-1]
        else:
            raise IndexError("Stack is empty")

    def size(self):
        return len(self.items)

if __name__ == '__main__':
    # Create a new stack
    stack = Stack()
    
    # Push elements onto the stack
    stack.push(10)
    stack.push(20)
    stack.push(30)
    
    # Get the top element (peek)
    print(stack.peek())  # Output: 30
    
    # Pop an element from the stack
    print(stack.pop())  # Output: 30
    
    # Check if the stack is empty
    print(stack.is_empty())  # Output: False
    
    # Get the current size of the stack
    print(stack.size())  # Output: 2

File: Stack/java/src/main/java/Stack.java
package src.main.java;
import java.util.ArrayList;
import java.util.EmptyStackException;

public class Stack<T> {
    private ArrayList<T> items;

    public Stack() {
        items = new ArrayList<>();
    }

    public boolean isEmpty() {
        return items.isEmpty();
    }

    public void push(T item) {
        items.add(item);
    }

    public T pop() {
        if (!isEmpty()) {
            return items.remove(items.size() - 1);
        } else {
            throw new EmptyStackException();
        }
    }

    public T peek() {
        if (!isEmpty()) {
            return items.get(items.size() - 1);
        } else {
            throw new EmptyStackException();
        }
    }

    public int size() {
        return items.size();
    }

    public static void main(String[] args) {
        // Create a new stack
        Stack<Integer> stack = new Stack<>();

        // Push elements onto the stack
        stack.push(10);
        stack.push(20);
        stack.push(30);

        // Get the top element (peek)
        System.out.println(stack.peek()); // Output: 30

        // Pop an element from the stack
        System.out.println(stack.pop()); // Output: 30

        // Check if the stack is empty
        System.out.println(stack.isEmpty()); // Output: false

        // Get the current size of the stack
        System.out.println(stack.size()); // Output: 2
    }
}
